functions/src/calculator/index.js

```diff
@@ -0,0 +1,47 @@
+import * as functions from 'firebase-functions'
+import * as admin from 'firebase-admin'
+import calculateResult from '../utils/calculateResult'
+import { to } from '../utils/async'
+
+/**
+ * @param req - Express HTTP Request
+ * @param res - Express HTTP Response
+ */
+async function calculatorRequest(req, res) {
+  const number1 = Number(req.body.number1)
+  const number2 = Number(req.body.number2)
+  const operation = req.body.operation
+  const result = calculateResult(number1, number2, operation)
+
+  const [rtdbError] = await to(
+    admin
+      .database()
+      .ref('/calculated')
+      .set(result)
+  )
+
+  const [firestoreError] = await to(
+    admin
+      .firestore()
+      .collection('calculated')
+      .doc('value')
+      .set({ result: result })
+  )
+
+  let htmlResponse = ''
+  if (rtdbError || firestoreError) {
+    htmlResponse = 'Error is storing the data'
+  } else {
+    htmlResponse = 'Data Successfully Stored'
+  }
+
+  res.writeHead(200, { 'Content-Type': 'text/html; charset=utf-8' })
```

Comment:
No change nessesary, but something to note is that we are currently returning html, but in some cases we will be responding with JSON.